### PR TITLE
Add pull tests from tri-adam

### PR DESCRIPTION
**Pull request by jscook2345**
**Description of the Pull Request (PR):**

Add some pull tests originally created by @tri-adam . Close #1793.

- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [X] I have tested this PR locally with a `make test`
- [X] This PR is against the project's master branch
- [ ] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [X] This PR is ready for review and/or merge

### CODE REVIEW

File: src/cmd/singularity/pull_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/pull_test.go]: LICENSE.md
or @jmstover get angry

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
small change

File: src/cmd/singularity/pull_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+	return exec.Command(cmdPath, argv...).CombinedOutput()
+}
+
+func TestPullForce(t *testing.T) {
+	test.EnsurePrivilege(t)
+
+	imagePath := "./ubuntu_17.10.sif"
+	defer os.Remove(imagePath)
+	if b, err := imagePull(pullOpts{}, "library://dtrudg/linux/ubuntu:17.10"); err != nil {
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/pull_test.go]: I recommend using `linux/alpine:3.7`

File: src/cmd/singularity/pull_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	argv = append(argv, sourceSpec)
+
+	return exec.Command(cmdPath, argv...).CombinedOutput()
+}
+
+func TestPullForce(t *testing.T) {
+	test.EnsurePrivilege(t)
+
+	imagePath := "./ubuntu_17.10.sif"
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/pull_test.go]: `alpine_3.7.sif`

File: src/cmd/singularity/pull_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	defer os.Remove(imagePath)
+	if b, err := imagePull(pullOpts{}, "library://dtrudg/linux/ubuntu:17.10"); err != nil {
+		t.Log(string(b))
+		t.Fatalf("unexpected failure: %v", err)
+	}
+	imageVerify(t, imagePath, false)
+
+	if b, err := imagePull(pullOpts{}, "library://dtrudg/linux/ubuntu:17.10"); err == nil {
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/pull_test.go]: I recommend using `linux/alpine:3.7`

File: src/cmd/singularity/pull_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		t.Log(string(b))
+		t.Fatalf("unexpected success")
+	}
+
+	if b, err := imagePull(pullOpts{force: true}, "library://dtrudg/linux/ubuntu:17.10"); err != nil {
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/pull_test.go]: same: I recommend using `linux/alpine:3.7`

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
small nitpick, that could save some seconds when testing

**Review by ArangoGutierrez — APPROVED** [review]

**Comment by ArangoGutierrez** [comment]
@tri-adam  RFR

**Comment by tri-adam** [comment]
I can't approve this (I wrote it) but one review is enough. Thanks gents.

**Pull request merged**